teuthology/dispatcher/supervisor.py

```diff
@@ -151,6 +153,45 @@ def run_job(job_config, teuth_bin_path, archive_dir, verbose):
     if 'targets' in job_config:
         unlock_targets(job_config)
 
+def mark_machine_down(machine_name, count):
+    teuthology.lock.ops.update_lock(machine_name,
+                                    description='reimage failed {0} times'.format(count),
+                                    status='down',
+                                    )
+    log.exception(
+        'Reimage failed {0} times ... marking machine down'.format(count)
+    )
```

Comment:
`log.exception` isn't really appropriate here; I think `log.error` is pretty sensible